pkg/malwarescan/malwarescan.go

```diff
@@ -34,14 +75,113 @@ type ScanError struct {
 
 // Client : Interface for the malwarescan api provided by SAP CP (see https://api.sap.com/api/MalwareScanAPI/overview)
 type Client interface {
+	// ScanAsync(candidate io.Reader) (*ScanResult, error)
+	ScanAsync(file string) (*ScanResult, error)
 	Scan(candidate io.Reader) (*ScanResult, error)
 	Info() (*Info, error)
 }
 
 // ClientImpl : Client implementation of the malwarescan api provided by SAP CP (see https://api.sap.com/api/MalwareScanAPI/overview)
 type ClientImpl struct {
-	HTTPClient piperhttp.Sender
-	Host       string
+	HTTPClient     piperhttp.Sender
+	Host           string
+	PollInterval   time.Duration
+	PollingTimeout time.Duration
+}
+
+// ScanAsync : Triggers an asynchronous malwarescan for the given content.
+func (c *ClientImpl) ScanAsync(file string) (*ScanResult, error) {
+	var scanResult ScanResult
+	var asyncUploadResult AsyncUploadResult
+
+	log.Entry().Debugf("Starting asynchronous scan")
+
+	// Without using a bytes.Reader for the request body, the net/http NewRequest
+	// method does not set Content-Length header and set the transfer to "chunked"
+	// which is not supported by the malware scan api.
+	fileHandle, err := os.ReadFile(file)
```

Comment:
Don't read the file into RAM...
The application will run out of memory eventually
 I also don't understand the comment.
 Why not just using `os.Open(file)` which returns an io.Reader that can be used as body for the http request?